### ISSUE TITLE
Case test being evaluated when it should not be

### ISSUE THREAD

**Issue by oznetmaster** [sebastienros/jint#535]
https://github.com/sebastienros/jint/blob/77ed14e82b4be972fdc8df9bafd56685096577e4/Jint/Runtime/StatementInterpreter.cs#L384

According to the ECMA standard, once a case test has been satisfied, subsequent case test clauses that are encountered (fallen through) are not to be evaluated.

The existing code evaluates every case test encountered.

**Comment by sebastienros** [comment]
Would you mind contributing a PR that add a test case which is failing?

**Comment by oznetmaster** [comment]
Apparently, there is no ECMA unit test which tests for this.

A test case would need to have CaseSelector which has side effects if evaluated, but which is a fall through CaseSelector from a previously matched case selector.

The fix is very simple.  else if (!hit) instead of else.

**Comment by sebastienros** [comment]
Exactly, there are no test cases for this, which is the exact reason I asked if you could contribute such a treat case ;) I would then see exactly what the issue is and fix it.

**Comment by sebastienros** [comment]
I get it now, you are just trolling, well done!

**Comment by oznetmaster** [comment]
No.  I have just completed porting this entire library to a very specialized Windows CE 7  platform (which only runs a subset of C# 3.0).  As such, I had to examine virtually line of code.  Since I believe in the code review process, I had the ECMA spec available as I did so, and this error in the existing code was one which seemed to have been overlooked, so I reported it to you,  What you do with that information is yours to decide.  I have no bandwidth to be a contributor to your project.
